docs/s3-bucket-versioning.md

```diff
@@ -0,0 +1,78 @@
+# **Steps to Create and Configure a Versioned S3 Bucket**
+
```

Comment:
Would be good to have a summary about what this doc is about here:
Something like:

This file provides a step-by-step guide for creating and configuring an Amazon S3 bucket with versioning and lifecycle management to optimize storage usage and costs.